start.py

```diff
@@ -558,40 +563,54 @@ def _command_condition(deployment, command_id, command_description):
     return not active and success
 
 
-# TODO: Get rid of the excess amount of code duplication between the following
-# function and _configure_kudu.
-def _configure_kafka(deployment, cluster, args):
+def _deactivate_service_parcel(deployment, cluster, product, version):
+    parcel_found_state = 'not_found'
     for parcel in deployment.cluster(DEFAULT_CLUSTER_NAME).parcels:
-        if parcel.product == 'KAFKA' and parcel.stage in 'ACTIVATED':
-            parcel_kafka_version = parcel.version.split('-')[0]
-            if parcel_kafka_version == args.kafka_version:
-                logger.info('Detected Kafka version matches specified version. Continuing ...')
+        if parcel.product == product and parcel.stage in 'ACTIVATED':
+            parcel_version = parcel.version.split('-')[0]
+            if parcel_version == version:
+                logger.info('Detected parcel version matches specified version. Continuing ...')
+                parcel_found_state = 'matched_active_version'
             else:
                 parcel.deactivate()
+                parcel_found_state = 'is_deactivated'
+            # Only one parcel can be activated at a time, so once one is found, our work is done.
+            break
+    return parcel_found_state
```

Comment:
I thought I had commented this before, but returning strings like this seems weird.

Reply:
Done.